WordPress/Classes/ViewRelated/Post/PostPreviewViewController.m

```diff
@@ -214,6 +214,27 @@ - (BOOL)webView:(UIWebView *)awebView
     if (navigationType == UIWebViewNavigationTypeFormSubmitted) {
         return NO;
     }
+    
+    // Something is causing our web view to try and load a different URL after the initial URL is already loading.
+    //
+    // Example of first URL: https://diegotest4.wordpress.com/?p=748
+    // Example of second URL: https://public-api.wordpress.com/wp-admin/rest-proxy/#https://diegotest4.wordpress.com
+    //
+    // The second URL is probably loaded by Javascript code or is part of a frame, as it's also loaded by Safari
+    // according to Charles.app (not just in our app in WPiOS).
+    //
+    // The problem is that our code is failing to filter this second URL, and allowing it to load.  This fix
+    // implements a final safeguard in the logic of this method, after all other checks are above pass.
+    //
+    // We won't allow a URL that's not the base document to load at this point.
+    // It could be worth our effort to try removing this check once we migrate our web view to WKWebView.
+    //
+    // More information and context on this issue can be found here: https://git.io/fj5Px
+    //
+    if (![request.URL isEqual:request.mainDocumentURL]) {
```

Comment:
@diegoreymendez I did some more tests and found that this prevents Youtube embeds from being loaded. 

The steps that I did:

1. Create a post in Calypso with a Youtube embed and some images. 
2. Publish it. 
3. View the post in WPiOS.

The images load fine but the Youtube embed didn't.  The post is here: https://yellingyaktest001.home.blog/2019/08/14/videos/. And this is what I see:

<img src="https://user-images.githubusercontent.com/198826/63036594-8f3df280-be7a-11e9-91b2-653de4e04f5f.PNG" width="320">

The Youtube embed shows up using the preview shown if I click on _Visit_ in the Reader.

I validated that the Youtube embed was canceled by adding a breakpoint on line 235. These URLs were ignored:

- `https://www.youtube.com/embed/oB6b5P6bcZg?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent`
- `https://widgets.wp.com/likes/master.html?ver=20190321#ver=20190321`

Reply:
@rachelmcr Would you mind testing Youtube embeds too? Maybe it's possible that it's just me experiencing this. 😬

Reply:
Awesome feedback.  I'll try to dig deeper into how to make this safe for us.  Eli proposed checking for the specific URL.

Reply:
Oh, good catch! I didn't think to test posts with embeds, but when I do there are a large number of embeds that never load (YouTube but also embeds like Slideshare, Spotify, Tumblr, VideoPress). I should note though that (I guess due to the original issue) most of these embeds also don't load for me in the production release.

Reply:
Thanks for that info @rachelmcr!